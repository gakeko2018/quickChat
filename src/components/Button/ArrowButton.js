import React from "react";
import {
  StyleSheet,
  Text,
  TextInput,
  KeyboardAvoidingView,
  TouchableOpacity,
  View
} from "react-native";
import { colors, margin } from "../../styles/base";

const ArrowButton = ({ text, style, onPress }) => {
  return (
    <View style={[styles.leaveButton, style]}>
      <View style={styles.leaveButtonLeft} />
      <View style={styles.leaveButtonRight}>
        <TouchableOpacity onPress={onPress}>
          <Text style={styles.leaveButtonText}> {text} </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  leaveButton: {
    flexDirection: "row"
  },
  leaveButtonLeft: {
    borderRightWidth: 10,
    borderRightColor: colors.tertiaryBackground,
    borderTopWidth: 15,
    borderTopColor: "transparent",
    borderBottomWidth: 15,
    borderBottomColor: "transparent",
    height: 0,
    width: 0,
    left: -10,
    top: 0,
    position: "absolute"
  },
  leaveButtonRight: {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: colors.tertiaryBackground,
    height: 30,
    width: 100,
    borderWidth: 1,
    borderColor: colors.tertiaryBackground,
    borderStyle: "solid"
  },
  leaveButtonText: {
    color: colors.primaryFontColor,
    fontSize: 20
  }
});

export default ArrowButton;
