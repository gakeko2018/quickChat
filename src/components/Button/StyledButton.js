import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import { colors, margin } from "../../styles/base";
import PropTypes from "prop-types";

const StyledButton = ({ text, style, onPress }) => {
  return (
    <TouchableOpacity style={[styles.button, style]} onPress={onPress}>
      <Text style={styles.buttonText}> {text} </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    margin: margin.sm,
    backgroundColor: colors.tertiaryBackground,
    borderRadius: 10,
    minHeight: 30
  },
  buttonText: {
    fontSize: 20,
    color: colors.primaryFontColor
  }
});

StyledButton.propTypes = {
  text: PropTypes.string,
  style: PropTypes.object,
  onPress: PropTypes.func
};

export default StyledButton;
