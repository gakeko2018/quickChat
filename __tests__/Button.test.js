import React from "react";
import { shallow } from "enzyme";
import ArrowButton from "../src/components/Button/ArrowButton";
import StyledButton from "../src/components/Button/StyledButton";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let wrapper2;

beforeEach(() => {
  wrapper = shallow(<ArrowButton text="ArrowButton" onPress={() => {}} />);
  wrapper2 = shallow(<StyledButton text="ArrowButton" onPress={() => {}} />);
});

describe("Buttons renders correctly", () => {
  it(" ArrowButton renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it(" StyledButton renders correctly", () => {
    expect(wrapper2).toMatchSnapshot();
  });
});
