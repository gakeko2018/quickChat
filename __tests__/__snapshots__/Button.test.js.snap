// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders correctly  ArrowButton renders correctly 1`] = `ShallowWrapper {}`;

exports[`Buttons renders correctly  StyledButton renders correctly 1`] = `ShallowWrapper {}`;
